authorize/evaluator/evaluator.go

```diff
@@ -134,28 +136,28 @@ func (e *Evaluator) Evaluate(ctx context.Context, req *Request) (*Result, error)
 		return nil, fmt.Errorf("error signing JWT: %w", err)
 	}
 
+	evalResult := &Result{
+		MatchingPolicy: getMatchingPolicy(res[0].Bindings.WithoutWildcards(), e.policies),
+		SignedJWT:      signedJWT,
+	}
+	evalResult.UserEmail, evalResult.UserGroups = e.UserInfo(req)
```

Comment:
This is not what I mean in previous review. What I mean is we should move the payload in `SignedJWT` to separated function, so there's not duplicated code. That said, currently:

```
signedJWT, err := e.SignedJWT(req)
if err != nil {
	return nil, fmt.Errorf("error signing JWT: %w", err)
}
```

becomes:

```
payload := jwtPayload(req)  // This will return a map[string]interface{}
signedJWT, err := e.SignedJWT(payload)
if err != nil {
	return nil, fmt.Errorf("error signing JWT: %w", err)
}

evalResult := &Result{
	MatchingPolicy: getMatchingPolicy(res[0].Bindings.WithoutWildcards(), e.policies),
	SignedJWT:      signedJWT,
}
if e, ok := payload["email"].(string); ok {
	evalResult.UserEmail = e
}
// Do the same for group
```